#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.09835915302414734 0.011286315239702225 -0.995086979165481 2.9141842357616303
  position -0.32496590695311056 0.2541762599509915 0.5368048218017067
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5.5 5.5
}
Robot {
  translation -2 0 0.0397031
  rotation 0.001214870299984249 0.00017388504293690775 -0.9999992469267897 -0.02882530717958609
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position 224.8127318172125
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint Solid {
        translation 0.05000825233443029 0.06000067095540662 7.612074135080459e-06
        rotation 0.6498807068480325 -0.5374269690988256 0.5374265714978668 1.989022090958409
        children [
          DEF Rueda Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "rueda1"
        boundingObject USE Rueda
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 173.6172631820924
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint Solid {
        translation 0.05001092510669232 -0.059999283239963785 8.675617962676796e-06
        rotation -0.2973226988783414 0.6751302550205651 -0.6751283962977455 3.7195811203135536
        children [
          USE Rueda
        ]
        name "rueda2"
        boundingObject USE Rueda
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 173.61727191872484
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "motor3"
        }
      ]
      endPoint Solid {
        translation -0.049990982090307994 -0.0600006734943581 6.80045476866652e-06
        rotation -0.29732597430555796 0.6751296004934318 -0.6751276083384754 3.7195871607636315
        children [
          USE Rueda
        ]
        name "rueda3"
        boundingObject USE Rueda
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 224.81274137329703
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "motor4"
        }
      ]
      endPoint Solid {
        translation -0.04998960928037487 0.059999282978267565 9.358991434998793e-06
        rotation 0.6498844130711621 -0.5374247722279122 0.5374242866143359 1.9890167781535844
        children [
          USE Rueda
        ]
        name "rueda4"
        boundingObject USE Rueda
        physics Physics {
        }
      }
    }
    DEF Cuerpo Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    InertialUnit {
      translation 0.1 0 0.05
      children [
        DEF IMU Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.02
            height 0.03
          }
        }
      ]
      name "imu"
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Box {
          size 0.05 0.05 0.05
        }
      }
      physics Physics {
      }
    }
    Lidar {
      translation 0 0 0.08
      children [
        DEF Lidar Shape {
          appearance PBRAppearance {
            baseColor 0.3 0.5 0.5
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      boundingObject USE Lidar
      physics Physics {
      }
      fieldOfView 1.8
      numberOfLayers 1
      maxRange 1.5
    }
    Display {
    }
    GPS {
      translation 0 0 0.05
      children [
        DEF GPS Shape {
          appearance PBRAppearance {
            baseColor 0.6 0.5 0.3
          }
          geometry Cylinder {
            height 0.05
            radius 0.02
          }
        }
      ]
      boundingObject USE GPS
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.1 0.02 0
      rotation 0 0 1 0.3
      children [
        DEF sensorDistancia Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "distIzq"
      boundingObject USE sensorDistancia
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.1 -0.02 0
      rotation 0 0 1 -0.3
      children [
        USE sensorDistancia
      ]
      name "distDer"
      boundingObject USE sensorDistancia
      physics Physics {
      }
    }
  ]
  boundingObject USE Cuerpo
  physics Physics {
  }
  controller "my_controller"
  window "<none>"
}
WoodenBox {
  translation 0.58 -0.06 0.3
}
